drivers/manual_hmis_data/app/views/manual_hmis_data/_form.haml

```diff
@@ -0,0 +1,4 @@
+- @fields.each do |key, value|
+  .row
+    .col-sm-12
+      = form.input key, label: key, as: value[:type], input_html: value[:html_input], collection: value[:collection] || [], include_blank: value[:include_blank] || true, required: value[:required] || false
```

Comment:
I _think_ this would be more readable as 
```
options = {input_html: {}, collection: [], include_blank: true, required: false}.merge(value)
= form.input key, label: key, **options
```

Reply:
(That's untested)

Reply:
I like the gist of the suggestion, but it needs a slight modification because we need to rename the `:type` key from value to `:as`. I'm thinking:

```
- defaults = {input_html: {}, collection: [], include_blank: true, required: false}
- @fields.each do |key, value|
  - options = defaults.merge(value)
  - options[:as] = options.delete(:type)
  .row
    .col-sm-12
      = form.input key, label: key, **options
```

Reply:
Looks good to me!